test/test_upload_mock.py

```diff
@@ -478,24 +462,24 @@ def generate_expected_output(
             "-f "
             '"{data_dir}/packages/arduino/hardware/samd/1.8.11/variants/arduino_zero/openocd_scripts/arduino_zero.cfg" '
             '-c "telnet_port disabled; program '
-            "{{build_dir}/{sketch_name}.ino.bin} verify reset 0x2000; "
-            'shutdown"\n',
+            "{bo}{build_dir}/{sketch_name}.ino.bin{bc} verify reset "
+            '0x2000; shutdown"\n',
             "linux": '"{data_dir}/packages/arduino/tools/openocd/0.10.0-arduino7/bin/openocd" '
             "-d2 -s "
             '"{data_dir}/packages/arduino/tools/openocd/0.10.0-arduino7/share/openocd/scripts/" '
             "-f "
             '"{data_dir}/packages/arduino/hardware/samd/1.8.11/variants/arduino_zero/openocd_scripts/arduino_zero.cfg" '
             '-c "telnet_port disabled; program '
-            "{{build_dir}/{sketch_name}.ino.bin} verify reset 0x2000; "
-            'shutdown"\n',
+            "{bo}{build_dir}/{sketch_name}.ino.bin{bc} verify reset "
+            '0x2000; shutdown"\n',
             "win32": '"{data_dir}/packages/arduino/tools/openocd/0.10.0-arduino7/bin/openocd.exe" '
             "-d2 -s "
             '"{data_dir}/packages/arduino/tools/openocd/0.10.0-arduino7/share/openocd/scripts/" '
             "-f "
             '"{data_dir}/packages/arduino/hardware/samd/1.8.11/variants/arduino_zero/openocd_scripts/arduino_zero.cfg" '
             '-c "telnet_port disabled; program '
-            "{{build_dir}/{sketch_name}.ino.bin} verify reset 0x2000; "
-            'shutdown"\n',
+            "{bo}{build_dir}/{sketch_name}.ino.bin{bc} verify reset "
```

Comment:
This is not the best approach I think, you can just use double braces to make `format()` work without failing.
So this string becomes `{{{build_dir}/{sketch_name}.ino.bin}} verify reset ` and you won't need `format(bo="{", bc="}")` anymore.